nixos/modules/misc/mindustry.nix

```
@@ -0,0 +1,41 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib; let
+  cfg = config.services.mindustry-server;
+in {
+  options = {
+    services.mindustry-server = {
+      enable = mkEnableOption "Mindustry server";
+
+      package = mkPackageOption pkgs "mindustry-server" {};
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Open ports in the firewall for the mindustry server";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.mindustry = {
+      enable = true;
+      description = "Mindustry server";
+      serviceConfig = {
+        Type = "simple";
+        ExecStart = "${pkgs.mindustry-server}/bin/mindustry-server host";
+        ExecStop = "pkill server-release.jar";
+      };
+      wantedBy = ["network.target"];
```

Usually, applications are started when systemd reaches `multi-user.target`, unless the application is actually needed to set up networking (i.e. a VPN server/client). In this case we can add an `after` dependency to make sure the application starts, once `network.target` is active.


```suggestion
      wantedBy = ["multi-user.target"];
      after = ["network.target"];
```